src/briefcase/platforms/android/gradle.py

```diff
@@ -110,11 +111,11 @@ def build_app(self, app: BaseConfig, **kwargs):
                 # Windows needs the full path to `gradlew`; macOS & Linux can find it
                 # via `./gradlew`. For simplicity of implementation, we always provide
                 # the full path.
-                [str(self.gradlew_path(app)), "assembleDebug"],
+                [os.fsdecode(self.gradlew_path(app)), "assembleDebug"],
                 env=self.android_sdk.env,
                 # Set working directory so gradle can use the app bundle path as its
                 # project root, i.e., to avoid 'Task assembleDebug not found'.
-                cwd=str(self.bundle_path(app)),
+                cwd=os.fsdecode(self.bundle_path(app)),
```

Comment:
Regarding `cwd`: Changed in version 3.7: cwd parameter accepts a path-like object on Windows. Shlould I remove os.fsdecode() ? 

Docs doesn't say explicitly that `subprocess.run` args can contain path-like object, but it says that: "The full function signature is largely the same as that of the Popen constructor - most of the arguments to this function are passed through to that interface." Popen args can contain path-like object. I removed `os.fsdecode` from L114, and test was successful.

Reply:
btw I removed all `os.fsdecode` from subprocess.run args and cwd, I will readd them if needed

Reply:
The implication of that documentation is that it won't work on *Windows* and Python 3.6, so while it will work everywhere else on 3.6, we need to retain the calls for Windows support. 

Interestingly, in my testing of `subprocess.run` on Windows with Python 3.6, `cwd` *doesn't* require the conversion, but it *is* required on the arguments; the [original 3.6 docs](https://docs.python.org/3.6/library/subprocess.html#subprocess.Popen) seem to suggest this is correct behavior for cwd, so I'm not sure what the reference in the current 3.9 docs are referring to. It works fine on Python 3.8 without the conversion in either place.

So - my inclination is to keep it for the args (or, add it back in) for the args, with a docstring indicating a cleanup when we drop 3.6.